### ISSUE TITLE
GW1 autologin dont work

### ISSUE THREAD

**Issue by SirWild** [Healix/Gw2Launcher#282]
Hey! 
I wanted to use your Launcher with GW1.
Set up everything and start the client - all fine, but there is no autologin..

![grafik](https://user-images.githubusercontent.com/101933432/230676944-bdda8856-576f-4772-8ffc-685f784243fe.png)

I also checked the checkbox, but when I press OK and check the settings again, it's not marked anymore.

**Comment by Healix** [comment]
It was broken in the last release version. Use the [beta version](https://github.com/Healix/Gw2Launcher/raw/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe).

**Comment by SirWild** [comment]
I used the beta version, that you have linked, but still had the same error.

**Comment by Healix** [comment]
What error? The problem in the release version was the email, password and character name fields could not be changed. That was fixed in the beta version.

To enable the automatic login, you have to enter the email, password and character name, which appear below the "enable automatic login" option when it's checked. If one of those is left empty, the option won't stay checked, since these are required by GW1.

**Comment by SirWild** [comment]
Thanks its working now! 
Can I use uMod with your launcher?

**Comment by Healix** [comment]
Place d3d9.dll from uMod in the folder where Gw.exe is located, then have uMod running in the background prior to launching the account. Once the account is launched, it should appear in uMod.

**Comment by SirWild** [comment]
Everything works! Thanks a lot!

**Comment by SirWild** [comment]
I have a new question: I rename the gw.exe and got an error:
![grafik](https://user-images.githubusercontent.com/101933432/232420405-25a7b83c-63bf-4bf9-888b-6abadb2db038.png)
How can I fix it?

**Comment by Healix** [comment]
You can hover over the failed status to see a reason. "Gw.exe" is based on the path supplied under Settings > Guild Wars 1. The same name is used for all copies, so if you were to rename 1 copy, it'll fail to launch that account because "Gw.exe" doesn't exist.

What are you wanting to do? I recently found out someone else was doing this because they needed a unique name per executable, so I'll eventually be adding a proper solution for this. As a workaround, you could create a link. For example, if you had C:\\GW1\\Gw1.exe, start a command prompt as an administrator and enter:

mklink "C:\\GW1\Gw.exe" "C:\\GW1\Gw1.exe"

This will cause Gw2Launcher to think it's launching "Gw.exe" but it'll end up as "Gw1.exe".

**Comment by SirWild** [comment]
yhea, I also need a unique name per executable.
However - it works with the mklink :)

I will write a batch for this and let it autostart. So it's fine for me. 
But of course a proper solution would be great.

**Comment by SirWild** [comment]
uhm i got another issue now - GWToolbox cant find the gw1 process anymore, when I start the account via GW2Launcher. 
When I start the account normaly (with its own process name) - GWToolbox can find it.

**Comment by Healix** [comment]
I tested it. Gw.exe renamed to Gw1.exe, then launched using a symlink Gw.exe. GWToolbox shows the error "couldn't find character name RVA". Is that what's happening?

It's caused by the symlink. Right click any account > Settings > General > Windows. Enable one of the taskbar grouping options. This will change how accounts are launched, which will avoid the problem.

If it's not detecting the client at all, run GWToolbox as an admin.

**Comment by SirWild** [comment]
yes! Thanks! Now its working.

**Comment by SirWild** [comment]
I try to setup GWToolbox as a Mod to autoinject it into the started Guild Wars 1 client. Can I use the .dll from GWToolbox somehow with GW2 Launcher?

**Comment by Healix** [comment]
If you're asking if Gw2Launcher can inject the DLL instead of GWToolbox.exe, no.

**Comment by SirWild** [comment]
[https://www.youtube.com/watch?t=335&v=rehKLa7VrjU&feature=youtu.be](url)
There is a possibility to load the dll from GWToolbox as a mod. Have times the video attached to it. .

Can GW2Launcher load a dll as a mod?

**Comment by Healix** [comment]
No. That's what injecting is. The video is using "GW Launcher" to inject the DLLs in the plugins folder into GW1. Injecting modifies a running program to force it to load an outside DLL, which typically gets flagged as a virus. For comparison, there are DLLs that GW1 will load itself, notably d3d9.dll, which you can simply drop into the folder where Gw.exe is located to have it load. 

GWToolbox requires injecting. Surprisingly, they don't have an automated option for it.
